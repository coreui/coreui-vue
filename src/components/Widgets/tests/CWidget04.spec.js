import { mount } from '@vue/test-utils'
import Component from '../CWidget04'

const ComponentName = 'CWidget04'
const defaultWrapper = mount(Component)
const customWrapper = mount(Component,{
  context:{
    props: {
      variant: 'twitter',
      rightHeader: 'rightHeader',
      leftHeader: 'leftHeader',
      rightFooter: 'rightFooter',
      leftFooter: 'leftFooter'
    }
  }
})

describe(ComponentName, () => {
  it('has a name', () => {
    expect(Component.name).toMatch(ComponentName)
  })
  it('is functional component', () => {
    expect(defaultWrapper.isFunctionalComponent).toBe(true)
  })
  it('correctly render component with default props', () => {
    expect(defaultWrapper.findAll('div').at(1).attributes().class).toMatch('brand-card-header bg-facebook')
    expect(defaultWrapper.findAll('div').at(4).text()).toMatch('rightHeader:string')
    expect(defaultWrapper.findAll('div').at(5).text()).toMatch('rightFooter:string')
    expect(defaultWrapper.findAll('div').at(7).text()).toMatch('leftHeader:string')
    expect(defaultWrapper.findAll('div').at(8).text()).toMatch('leftFooter:string')
    expect(defaultWrapper.find('i').attributes().class).toMatch('fa fa-facebook')
  })
  it('matches default props snapshot', () => {
    expect(defaultWrapper.element).toMatchSnapshot()
  })
  it('correctly render component with custom props', () => {
    expect(customWrapper.findAll('div').at(1).attributes().class).toMatch('brand-card-header bg-twitter')
    expect(customWrapper.findAll('div').at(4).text()).toMatch('rightHeader')
    expect(customWrapper.findAll('div').at(5).text()).toMatch('rightFooter')
    expect(customWrapper.findAll('div').at(7).text()).toMatch('leftHeader')
    expect(customWrapper.findAll('div').at(8).text()).toMatch('leftFooter')
    expect(customWrapper.find('i').attributes().class).toMatch('fa fa-twitter')
  })
  it('matches custom props snapshot', () => {
    expect(customWrapper.element).toMatchSnapshot()
  })
})
