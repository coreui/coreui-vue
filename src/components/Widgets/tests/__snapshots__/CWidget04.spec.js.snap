// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CWidget04 matches custom props snapshot 1`] = `
<div
  class="brand-card"
>
  <div
    class="brand-card-header bg-twitter"
  >
    <i
      class="fa fa-twitter"
    />
     
  </div>
   
  <div
    class="brand-card-body"
  >
    <div>
      <div
        class="text-value"
      >
        rightHeader
      </div>
       
      <div
        class="text-uppercase text-muted small"
      >
        rightFooter
      </div>
    </div>
     
    <div>
      <div
        class="text-value"
      >
        leftHeader
      </div>
       
      <div
        class="text-uppercase text-muted small"
      >
        leftFooter
      </div>
    </div>
  </div>
</div>
`;

exports[`CWidget04 matches default props snapshot 1`] = `
<div
  class="brand-card"
>
  <div
    class="brand-card-header bg-facebook"
  >
    <i
      class="fa fa-facebook"
    />
     
  </div>
   
  <div
    class="brand-card-body"
  >
    <div>
      <div
        class="text-value"
      >
        rightHeader:string
      </div>
       
      <div
        class="text-uppercase text-muted small"
      >
        rightFooter:string
      </div>
    </div>
     
    <div>
      <div
        class="text-value"
      >
        leftHeader:string
      </div>
       
      <div
        class="text-uppercase text-muted small"
      >
        leftFooter:string
      </div>
    </div>
  </div>
</div>
`;
